presto-hive/src/test/java/com/facebook/presto/hive/TestHiveLogicalPlanner.java

```diff
@@ -953,6 +963,160 @@ public void testPushdownSubfieldsAndJoinReordering()
+        Session legacyUnnest = Session.builder(withParquetDereferencePushDownEnabled())
+                .setSystemProperty("legacy_unnest", "true")
+                .build();
+        assertParquetDereferencePushDown(legacyUnnest, "SELECT t.y.a, t.y.d.d1, x.a FROM test_pushdown_nestedcolumn_parquet CROSS JOIN UNNEST(y) as t(y)",
+                "test_pushdown_nestedcolumn_parquet",
+                nestedColumnMap("x.a"));
+
+        assertParquetDereferencePushDown(legacyUnnest, "SELECT t.*, x.a FROM test_pushdown_nestedcolumn_parquet CROSS JOIN UNNEST(y) as t(y)",
+                "test_pushdown_nestedcolumn_parquet",
+                nestedColumnMap("x.a"));
+
+        assertParquetDereferencePushDown(legacyUnnest, "SELECT id, x.a FROM test_pushdown_nestedcolumn_parquet CROSS JOIN UNNEST(y) as t(y)",
+                "test_pushdown_nestedcolumn_parquet",
+                nestedColumnMap("x.a"));
+
+        // Case sensitivity
+        assertParquetDereferencePushDown("SELECT x.a, x.b, x.A + 2 FROM test_pushdown_nestedcolumn_parquet WHERE x.B LIKE 'abc%'",
+                "test_pushdown_nestedcolumn_parquet",
+                nestedColumnMap("x.a", "x.b"));
+
+        // No pass-through nested column pruning
+        assertParquetDereferencePushDown("SELECT id, min(x.d).d1 FROM test_pushdown_nestedcolumn_parquet GROUP BY 1",
+                "test_pushdown_nestedcolumn_parquet",
+                nestedColumnMap("x.d"));
+
+        assertParquetDereferencePushDown("SELECT id, min(x.d).d1, min(x.d.d2) FROM test_pushdown_nestedcolumn_parquet GROUP BY 1",
+                "test_pushdown_nestedcolumn_parquet",
+                nestedColumnMap("x.d"));
+
+        // Test pushdown of filters on dereference columns
+        assertParquetDereferencePushDown("SELECT id, x.d.d1 FROM test_pushdown_nestedcolumn_parquet WHERE x.d.d1 = 1",
+                "test_pushdown_nestedcolumn_parquet",
+                nestedColumnMap("x.d.d1"),
+                withColumnDomains(ImmutableMap.of(
+                        pushdownColumnNameForSubfield(nestedColumn("x.d.d1")), singleValue(BIGINT, 1L))),
+                ImmutableSet.of(pushdownColumnNameForSubfield(nestedColumn("x.d.d1"))),
+                TRUE_CONSTANT);
+
+        assertParquetDereferencePushDown("SELECT id, x.d.d1 FROM test_pushdown_nestedcolumn_parquet WHERE x.d.d1 = 1 and x.d.d2 = 5.0",
+                "test_pushdown_nestedcolumn_parquet",
+                nestedColumnMap("x.d.d1", "x.d.d2"),
+                withColumnDomains(ImmutableMap.of(
+                        pushdownColumnNameForSubfield(nestedColumn("x.d.d1")), singleValue(BIGINT, 1L),
+                        pushdownColumnNameForSubfield(nestedColumn("x.d.d2")), singleValue(DOUBLE, 5.0))),
+                ImmutableSet.of(
+                        pushdownColumnNameForSubfield(nestedColumn("x.d.d1")),
+                        pushdownColumnNameForSubfield(nestedColumn("x.d.d2"))),
+                TRUE_CONSTANT);
```

Comment:
Make sure all these functions are with one parameter per line with the first line empty.